src/configurations.jl

```diff
@@ -871,6 +873,38 @@ for O in [Orbital, RelativisticOrbital]
     @eval get_noble_gas(::Type{<:$O}, k) = noble_gases_configurations[$O][k]
 end
 
+"""
+    nrconfiguration(c::Configuration{<:RelativisticOrbital}) -> Configuration{<:Orbital}
+
+Reduces a relativistic configuration down to the corresponding non-relativistic configuration.
+
+```jldoctest
+julia> c = rc"1s2 2p-2 2s 2p2 3s2 3p-"
+1s² 2s 2p⁻² 2p² 3s² 3p⁻
+
+julia> nrconfiguration(c)
+1s² 2s 2p⁴ 3s² 3p
+```
+"""
+function nrconfiguration(c::Configuration{<:RelativisticOrbital})
+    mq = Union{mqtype.(c.orbitals)...}
+    nrorbitals, nroccupancies, nrstates = Orbital{<:mq}[], Int[], Symbol[]
+    for (orbital, occupancy, state) in c
+        nrorbital = Orbital(orbital.n, orbital.ℓ)
+        nridx = findfirst(isequal(nrorbital), nrorbitals)
+        if isnothing(nridx)
+            push!(nrorbitals, nrorbital)
+            push!(nroccupancies, occupancy)
+            push!(nrstates, state)
+        else
+            @assert nrorbitals[nridx] == nrorbital
```

Comment:
Is this necessary? I.e. when would the assertion fail? I thought `findfirst` if not `isnothing` would return an index of an element that fulfills the predicate?

Reply:
Nope, you're right. Just an accidental leftover from development. Will remove.